Sources/Zip/Zip.swift

```diff
@@ -146,8 +146,8 @@ public class Zip {
             }
 
             var writeBytes: UInt64 = 0
-            while let filePointer = fopen(fullPath, "wb") {
-                defer { fclose(filePointer) }
+            let filePointer: UnsafeMutablePointer<FILE>? = fopen(fullPath, "wb")
+            while let filePointer {
                 let readBytes = unzReadCurrentFile(zip, &buffer, bufferSize)
                 if readBytes > 0 {
```

Comment:
Not part of the current changes but it could be re-written to:
```swift
guard readBytes > 0 else {
    break
}
guard fwrite(buffer, Int(readBytes), 1, filePointer) == 1 else {
    throw ZipError.unzipFail
}
writeBytes += UInt64(readBytes)
```
(I know this is just the original code that was there)